### PR TITLE
feature: add Navbar component

### PR DESCRIPTION
**Pull request by tbor00**
## Summary

The new Navbar component and its components are designed to give you more freedom to create custom navigation bars.

## Task

[CD-559](https://dd360.atlassian.net/browse/CD-559)

## How did you test this change?

All tests was passed with 100% of coverage ✅

### CODE REVIEW

**Comment by tbor00** [comment]
@AntonioFeregrino feel free to try with the codesandbox compilation 👍🏻

File: src/components/Dropdown/Dropdown.test.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    })
+
+    afterEach(() => cleanup())
+
+    it('should be render', () => {
+        const { container } = renderResult
+        expect(container).toBeDefined()
+    })
+
+    it('porps: disableAnimation', () => {
```
Comment (lautarovilas) [inline src/components/Dropdown/Dropdown.test.tsx:26]:
```suggestion
    it('props: disableAnimation', () => {
```

File: src/components/Navbar/Language.tsx [hunk 2]
```diff
@@ -12,33 +13,36 @@ const langs = [
...
-    }
+const langName: { [key: string]: string } = {
+    English: 'ENG',
+    Spanish: 'ESP'
+}
+
+export function Language({ isNavbar, defaultLanguage, changeLanguage, getLangName, ...props }: LanguageProps) {
+    const [language, setLanguage] = useState(defaultLanguage || 'es')
 
     const toggleLanguage = () => {
```
Comment (hasandd3) [inline src/components/Navbar/Language.tsx:31]:
```tsx
const toggleLanguage = useCallback(() => {
    const lang = language === 'es' ? 'en' : 'es'
    localStorage.setItem('language', lang)
    changeLanguage && changeLanguage(lang)
    setLanguage(lang)
}, [changeLanguage])
```

File: src/components/Navbar/Navbar.test.tsx [hunk 3]
```diff
@@ -1,11 +1,95 @@
 import { it, describe } from 'vitest'
-import { render } from '@testing-library/react'
-
-import { Navbar } from './Navbar'
+import { cleanup, fireEvent, render, RenderResult } from '@testing-library/react'
+import { getActiveVariants } from './NavbarLink'
+import { getActiveItemVariants } from './NavbarItem'
+import NavbarWrapper from './NavbarWrapper.mock'
 
 describe('<Navbar/>', () => {
```
Comment (hasandd3) [inline src/components/Navbar/Navbar.test.tsx:7]: <Navbar />

File: src/components/Navbar/Language.tsx [hunk 4]
```diff
@@ -12,33 +13,36 @@ const langs = [
...
+    defaultLanguage?: 'en' | 'es'
 }
 
-export function Language({ isNavbar, changeLanguage, getLangName }: Props) {
-    const [language, setLanguage] = useState(localStorage.getItem('language') ?? 'es')
-    const langName: { [key: string]: string } = {
-        English: 'ENG',
-        Spanish: 'ESP'
-    }
+const langName: { [key: string]: string } = {
```
[the remark this answers is not in the document]
Comment (AntonioFeregrino) [reply src/components/Navbar/Language.tsx:23]: langName is already defined outside the function, it shouldn't be redefined on every render

**Review by arielsrodriguez — APPROVED** [review]

**Comment by tbor00** [comment]
This PR have conflicts, @AntonioFeregrino  you can resolve?

**Comment by tbor00** [comment]
@AntonioFeregrino are you going to resume this PR? 👀

**Pull request merged**